project-icp/bee-pollinator-app
# Pan & zoom map from location search and apiary card marker

## Overview

The client requested additional map interactions. Previously the apiary cards reflected map markers but otherwise didn't interact with the map; location search panned but didn't zoom the map. The added interaction is clicking the apiary marker (A, B, C...AA) and location search both pan & zoom the map down to the coordinate location at map zoom.

To do this I cleaned up the map view handling. The redux state already saved the map center and map zoom, in some cases, but now it is the source of truth for setting the map view. 

Connects #531 

### Demo
![marker](https://user-images.githubusercontent.com/10568752/67230188-cf859800-f40a-11e9-9e65-8d813889a0a5.gif)


![zoomToLocation](https://user-images.githubusercontent.com/10568752/67230189-cf859800-f40a-11e9-806b-bb25f5f09f6d.gif)

### Notes

You might be thinking we could not set `zoomToResult` to true in the geocoder options; unfortunately no, because of a conflicting bounding box bug already documented as a comment in `Map.jsx`

## Testing Instructions

`./scripts/beekeepers.sh start`
Open up the dev tools and see the redux logs expected actions, no duplicates
Create some apiaries and click their letter markers. Zoom out and/or pan away and redo this.
Search addresses in the geocoder and select.
